pkg/reconciler/githubsource/webhook_client.go

```diff
@@ -36,16 +37,26 @@ type webhookOptions struct {
 }
 
 type webhookClient interface {
-	Create(ctx context.Context, options *webhookOptions) (string, error)
-	Delete(ctx context.Context, options *webhookOptions, hookID string) error
+	Create(ctx context.Context, options *webhookOptions, alternateGitHubAPIURL string) (string, error)
+	Delete(ctx context.Context, options *webhookOptions, hookID, alternateGitHubAPIURL string) error
 }
 
 type gitHubWebhookClient struct{}
 
-func (client gitHubWebhookClient) Create(ctx context.Context, options *webhookOptions) (string, error) {
+func (client gitHubWebhookClient) Create(ctx context.Context, options *webhookOptions, alternateGitHubAPIURL string) (string, error) {
 	logger := logging.FromContext(ctx)
 
 	ghClient := client.createGitHubClient(ctx, options)
+	if alternateGitHubAPIURL != "" {
+		//This is to support GitHub Enterprise, this might be something like https://github.company.com/api/v3/
+		baseURL, err := url.Parse(alternateGitHubAPIURL)
+		if err != nil {
+			logger.Infof("failed to create webhook - error occured parsing githubAPIURL %s, error was %v", alternateGitHubAPIURL, err)
+			return "", fmt.Errorf("error occured parsing githubAPIURL: %v", err)
+		}
+		ghClient.BaseURL = baseURL
```

Comment:
You can assign directly to `ghClient.BaseURL` above, because you'll throw away the client on error return.

Reply:
Changed in both Create and Delete - I think I ended up in this state as I hadn't moved the declaration of err in the Create method ..... but I've moved that now so both are more consistent.